### PR TITLE
Update winit

### PR DESCRIPTION
**Pull request by Ralith**
Builds, but still needs manual testing.

### CODE REVIEW

**Comment by patowen** [comment]
I was able to run this branch successfully on my Windows machine, and all the buttons still seem to work as expected.

File: client/src/graphics/window.rs [hunk 1]
```diff
@@ -111,194 +104,167 @@ impl Window {
...
-                                0.0,
-                                0.0,
-                                2.0 * (anticlockwise as u8 as f32 - clockwise as u8 as f32)
-                                    * dt.as_secs_f32(),
-                            );
-
-                            sim.step(dt, &mut self.net);
-                            last_frame = this_frame;
-                        }
+    /// Run the event loop until process exit
```
Comment (patowen) [inline client/src/graphics/window.rs:134]: This comment is now outdated.

File: client/src/graphics/window.rs [hunk 2]
```diff
@@ -707,3 +673,32 @@ struct Frame {
...
+    forward: bool,
+    back: bool,
+    left: bool,
+    right: bool,
+    up: bool,
+    down: bool,
+    jump: bool,
+    clockwise: bool,
+    anticlockwise: bool,
+    last_frame: Option<Instant>,
```
Comment (patowen) [inline client/src/graphics/window.rs:688]: Out of all fields, this one seems the most out of place in the `InputState` struct, but at the same time, it does provide context for the rest of the fields, and in the context of a simulation, "amount of time passed" is indeed one of the inputs, so I don't think moving this field somewhere else would necessarily be any better.

**Review by patowen — APPROVED** [review]
Looks good to me! I left a few comments for minor things. Thanks for working on this, as it would have been very tricky for me to do.

**Review by patowen — APPROVED** [review]

File: client/src/graphics/window.rs (hunk 2 above)
Comment (Ralith) [reply client/src/graphics/window.rs]: Eh, lifted it out.

**Pull request merged**